engine/src/main/scala/cromwell/engine/workflow/tokens/JobExecutionTokenDispenserActor.scala

```diff
@@ -123,15 +155,39 @@ class JobExecutionTokenDispenserActor(override val serviceRegistryActor: ActorRe
     context.unwatch(terminee)
     ()
   }
+
+  def tokenDispenserState: JsObject = {
+    val queueStates = JsArray(tokenQueues.toVector.map { case (tokenType, queue) =>
+      JsObject(Map(
+        "token type" -> JsString(s"BACKEND=${tokenType.backend}/TOKENLIMIT=${tokenType.maxPoolSize}/HOGFACTOR=${tokenType.hogFactor}"),
+        "queue state" -> queue.statusPrint
+      ))
+    })
+
+    JsObject(Map("Token Dispenser state" -> JsObject(Map(
+      "queues" -> queueStates,
+      "pointer" -> JsNumber(currentTokenQueuePointer),
+      "total token assignments" -> JsNumber(tokenAssignments.size)
+    ))))
+  }
+
+  private def logTokenAllocation(someInterval: FiniteDuration): Unit = {
+    log.info(tokenDispenserState.prettyPrint)
```

Comment:
We're slowly migrating away from spray, at a snail's pace 🐌

Can you switch this to circe? TOL we should be passing around case classes and not Json representations. But that could be a bigger fix later.

If one doesn't want to use case classes plus circe encoders, there are still ways of using [`Json.fromXyz()`](https://www.scala-exercises.org/circe/Json) to build up and pass around circe json.

Reply:
Yeah, case classes is definitely the right thing to use here, I just got lazy about defining an ADT schema first.